presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/TimeUuidFunctions.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.cassandra;
+
+import com.datastax.driver.core.utils.UUIDs;
+import io.airlift.slice.Slice;
+import io.prestosql.spi.function.Description;
+import io.prestosql.spi.function.LiteralParameters;
+import io.prestosql.spi.function.ScalarFunction;
+import io.prestosql.spi.function.SqlType;
+
+import java.util.UUID;
+
+import static com.datastax.driver.core.utils.UUIDs.unixTimestamp;
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.prestosql.spi.type.DateTimeEncoding.packDateTimeWithZone;
+import static io.prestosql.spi.type.DateTimeEncoding.unpackMillisUtc;
+import static io.prestosql.spi.type.TimeZoneKey.UTC_KEY;
+
+public final class TimeUuidFunctions
+{
+    private TimeUuidFunctions() {}
+
+    @Description("Cassandra minTimeuuid function")
+    @ScalarFunction
+    @LiteralParameters("p")
+    @SqlType("varchar(36)")
+    public static Slice minTimeuuid(@SqlType("timestamp(p) with time zone") long timestamp)
+    {
+        long epochMillis = unpackMillisUtc(timestamp);
+        return utf8Slice(UUIDs.startOf(epochMillis).toString());
+    }
+
+    @Description("Cassandra maxTimeuuid function")
+    @ScalarFunction
+    @LiteralParameters("p")
+    @SqlType("varchar(36)")
+    public static Slice maxTimeuuid(@SqlType("timestamp(p) with time zone") long timestamp)
```

Comment:
This will work with p <= 3 only. For p > 3, it will fail because there's no version that takes LongTimestampWithTimeZone.  I have a couple of comments:
* Do we want to support p > 3? If so, what would be the semantics of such operation?
* If we don't want to support it, maybe add the method regardless and make it fail with a descriptive message.

Reply:
@martint I think we don't need to support it because Cassandra cannot represent such digit. Let me update to deny p > 3.